sdk/daml-lf/validation/src/main/scala/com/daml/lf/validation/Upgrading.scala

```diff
@@ -295,24 +304,48 @@ object TypecheckUpgrades {
     Try(t.map(f(_).get).toSeq)
 
   def typecheckUpgrades(
-      present: (Ref.PackageId, Ast.Package),
+      present: (
+          Ref.PackageId,
+          Ast.Package,
+          Map[Ref.PackageId, (Ref.PackageName, Ref.PackageVersion)],
+      ),
       pastPackageId: Ref.PackageId,
-      mbPastPkg: Option[Ast.Package],
+      mbPastPkg: Option[(Ast.Package, Map[Ref.PackageId, (Ref.PackageName, Ref.PackageVersion)])],
   ): Try[Unit] = {
     mbPastPkg match {
       case None =>
         fail(UpgradeError.CouldNotResolveUpgradedPackageId(Upgrading(pastPackageId, present._1)));
-      case Some(pastPkg) =>
-        val tc = TypecheckUpgrades(Upgrading((pastPackageId, pastPkg), present))
+      case Some((pastPkg, pastPkgDeps)) =>
+        val tc = TypecheckUpgrades(Upgrading((pastPackageId, pastPkg, pastPkgDeps), present))
         tc.check()
     }
   }
+
+  def typecheckUpgrades(
```

Comment:
We add this function with the previous interface that didn't take dependencies in order to be able to merge the first part of this PR and build canton against it in step 1 of the merge dance. Part 1: https://github.com/digital-asset/daml/pull/19274

We remove this function in part 3: https://github.com/digital-asset/daml/pull/19275